drivers/sensor/lps2xdf/CMakeLists.txt

```diff
@@ -1,12 +1,15 @@
 # ST Microelectronics LPS22DF pressure and temperature sensor
 #
 # Copyright (c) 2023 STMicroelectronics
+# Copyright (c) 2023 PHYTEC Messtechnik GmbH
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 zephyr_library()
 
+zephyr_library_sources(lps2xdf.c)
 zephyr_library_sources(lps22df.c)
-zephyr_library_sources_ifdef(CONFIG_LPS22DF_TRIGGER    lps22df_trigger.c)
+zephyr_library_sources(lps28dfw.c)
```

Comment:
Fine, but I would use something like 
```zephyr_library_sources_ifdef(DT_HAS_LPS22DF_ENABLED, lps22df.c)```

I hope it is working

Reply:
It works now with `zephyr_library_sources_ifdef(CONFIG_DT_HAS_LPS22DF_ENABLED, lps22df.c)`. I like this solution, it cleaner than preprocessor macros.

I even tried this earlier without the `CONFIG_` prefix and got confused by thinking that the devicetree macros get resolved after CMake configuration.

Reply:
Yeah I think the one with `CONFIG_DT_HAS_LPS22DF_ENABLED`, you wanna be careful with `DT_HAS_...` in cmake files, make sure it's behind something that can be switched off so you can still define the device in the tree and disable the driver, but I think it's fine in this case since the whole file is behind `CONFIG_LPS2XDF`. Good stuff.